### PR TITLE
#17698 Set Application Name property for Db2

### LINKED ISSUE #17698: Set Application Name parameter in Db2
**Is your feature request related to a problem? Please describe.**
DBeaver does not sets APPLICATION_NAME parameter and it is so displayed "db2jcc_application" as it is displayed for all applications that do not set application name. In many tools it is difficult to distinguish which of database connection is DBeaver and which one is from other applications that also do not have set the name.

**Describe the solution you'd like**
Db2 database administrator frequently checks for applications that causes trouble on database with command:
`db2 list applications`
and DBeaver is in Application Name listed as "db2jcc_application", which is default name if no name is set. See "db2jcc_applica" bellow:

```
Auth Id  Application    Appl.      Application Id                    DB
         Name           Handle                                       Name
-------- -------------- ---------- --------------------------------- -------
U1033704 db2jcc_applica 2019       192.168.3.15.64789.220920100224   DBTEST
U3470252 DataGrip       2020       192.168.14.233.66823.22092013233  DBTEST

```
But as it can be seen DataGrip database tool properly sets the Application Name (last line).

If I dig it deeper into Db2 monitoring:
`SELECT APPLICATION_NAME, CLIENT_APPLNAME FROM TABLE(MON_GET_CONNECTION(NULL, -2)) WHERE CLIENT_APPLNAME IS NOT NULL ORDER BY CLIENT_APPLNAME`

It displays:
```
APPLICATION_NAME    CLIENT_APPLNAME
------------------- ---------------------
db2jcc_application  DBeaver 22.2.0 - Main
DataGrip            DataGrip 2022.2.4
```

As it seen DBeaver correctly sets CLIENT_APPLNAME info.

But "Application Name" is used in many tools as the only column displayed and there is no CLIENT_APPLNAME available. It is so impossible to see which of database connections are from DBeaver.

My suggestion is: In DBeaver set APPLICATION_NAME to "DBeaver" without program version, just like DataGrip does.
Data in CLIENT_APPLNAME is just fine and should not be changed.

Bellow is official IBM site how to set APPLICATION_NAME:
https://www.ibm.com/support/pages/name-db2-jdbc-application-appears-db2jccapplication-how-change-it

### PR DESCRIPTION
**Pull request by E1izabeth**
This parameter should be set to the product name (DBeaver) now.

image

`SELECT APPLICATION_NAME, CLIENT_APPLNAME FROM TABLE(MON_GET_CONNECTION(NULL, -2)) WHERE CLIENT_APPLNAME IS NOT NULL ORDER BY CLIENT_APPLNAME;`

### CODE REVIEW

**Review by LonwoLonwo — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.ext.db2/src/org/jkiss/dbeaver/ext/db2/model/DB2DataSource.java [hunk 1]
```diff
@@ -75,47 +75,52 @@ public class DB2DataSource extends JDBCDataSource implements DBCQueryPlanner, IA
...
+    private List<DB2XMLString> listXMLStrings;
 
-    private DB2CurrentUserPrivileges                             db2CurrentUserPrivileges;
+    private DB2CurrentUserPrivileges db2CurrentUserPrivileges;
 
-    private String                                               schemaForExplainTables;
+    private String schemaForExplainTables;
 
-    private Double                                               version;                                                                                                                  // Database
+    private Double version;
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.db2/src/org/jkiss/dbeaver/ext/db2/model/DB2DataSource.java:123]: I don't think these changes were intentional

File: plugins/org.jkiss.dbeaver.ext.db2/src/org/jkiss/dbeaver/ext/db2/model/DB2DataSource.java [hunk 2]
```diff
@@ -273,6 +286,7 @@ protected Connection openConnection(@NotNull DBRProgressMonitor monitor, @Nullab
                     CommonUtils.truncateString(DBUtils.getClientApplicationName(getContainer(), context, purpose), 255));
             } catch (Throwable e) {
                 // just ignore
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.db2/src/org/jkiss/dbeaver/ext/db2/model/DB2DataSource.java]:
```suggestion
```

File: plugins/org.jkiss.dbeaver.ext.db2/src/org/jkiss/dbeaver/ext/db2/model/DB2DataSource.java (hunk 1 above)
Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.ext.db2/src/org/jkiss/dbeaver/ext/db2/model/DB2DataSource.java:123]: I intentionally changed it.

Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.ext.db2/src/org/jkiss/dbeaver/ext/db2/model/DB2DataSource.java:123]: I found weird such formatting

**Review by ShadelessFox — APPROVED** [review]

**Comment by Matvey16** [comment]
Verified

**Pull request merged**